index.js

```diff
@@ -61,9 +58,23 @@ pluginRegistry
     }
 
     var taskDefinition = plugin({
+      getDependency: function getContextDependency(name) {
+        switch(name) {
+          case 'hr':
+            return require('./lib/util/hr');
+          case 'gulp':
+            return require('gulp');
+          case 'runSequence':
+            return require('run-sequence');
+          case 'taskYargsRun':
+            return taskYargsRun;
+          default:
+            throw new Error('Context dependency ' + name + ' is unknown');
+        }
+      },
       taskYargsRun: taskYargsRun,
-      gulp: gulp,
-      runSequence: runSequence
+      gulp: require('gulp'),
+      runSequence: require('run-sequence')
```

Comment:
all these dependencies need not be defined here, and only within `getDependency`, **after** all tasks have been converted